src/Npgsql/NpgsqlConnector.cs

```diff
@@ -674,14 +674,23 @@ async Task RawOpen(NpgsqlTimeout timeout, bool async, CancellationToken cancella
                         break;
                     case 'S':
                         var clientCertificates = new X509Certificate2Collection();
-                        var certPath = Settings.ClientCertificate ?? PostgresEnvironment.SslCert;
-
-                        if (certPath is null && PostgresEnvironment.SslCertDefault is string certPathDefault)
-                            certPath = certPathDefault;
+                        var certPath = Settings.ClientCertificate ?? PostgresEnvironment.SslCert ?? PostgresEnvironment.SslCertDefault;
 
                         if (certPath != null)
                         {
-                            cert = new X509Certificate2(certPath, Settings.ClientCertificateKey ?? PostgresEnvironment.SslKey);
+                            var password = Settings.ClientCertificateKeyPassword;
+#if NET5_0
+                            if (Path.GetExtension(certPath).ToUpperInvariant() != ".PFX")
```

Comment:
Is it a good idea to check the file extension to determine the certificate type? Is there a way we can check on the contents or similar instead?

Reply:
![image](https://user-images.githubusercontent.com/14861420/111030031-1ebdd580-8411-11eb-8f7d-d16169057c0d.png)

![image](https://user-images.githubusercontent.com/14861420/111030049-3c8b3a80-8411-11eb-8c36-c0c3bc67edca.png)

We technically can check with regex for the
```
-----BEGIN CERTIFICATE-----
...
...
...
-----END CERTIFICATE-----
```

Reply:
Yeah, I don't know. I guess we can go with the extension and see if users complain?

Reply:
Yeah, it should be pretty easy to change if there will be any problems.